pkg/graph/resolvers/system_intake_status_admin_test.go

```diff
@@ -0,0 +1,387 @@
+				DecisionState: models.SIDSNoGovernance,
+				State:         models.SystemIntakeStateCLOSED,
+			},
+			expectedStatus: models.SISANoGovernance,
+			expectError:    false,
+		},
+		{
+			testCase: "Decision Not Approved, open",
+			intake: models.SystemIntake{
+				Step:          models.SystemIntakeStepDECISION,
+				DecisionState: models.SIDSNotApproved,
+				State:         models.SystemIntakeStateOPEN,
+			},
+			expectedStatus: models.SISANotApproved,
+			expectError:    false,
+		},
+		{
+			testCase: "Decision Not Approved, closed",
+			intake: models.SystemIntake{
+				Step:          models.SystemIntakeStepDECISION,
+				DecisionState: models.SIDSNotApproved,
+				State:         models.SystemIntakeStateCLOSED,
+			},
+			expectedStatus: models.SISANotApproved,
+			expectError:    false,
+		},
+		// These cases should not be present in the actual app, but they are possible edge cases
+		{
+			testCase: "Decision No Decision, closed",
+			intake: models.SystemIntake{
+				Step:          models.SystemIntakeStepDECISION,
+				DecisionState: models.SIDSNoDecision,
+				State:         models.SystemIntakeStateCLOSED,
+			},
+			expectedStatus: models.SISAClosed,
+			expectError:    false,
+		},
+		// This case will be handled in application code to ensure this doesn't occur. If needed, further logic can be created to enforce it, such as a check constraint in the DB
+		{
+			testCase: "Decision No Decision, closed",
+			intake: models.SystemIntake{
+				Step:          models.SystemIntakeStepDECISION,
+				DecisionState: models.SIDSNoDecision,
+				State:         models.SystemIntakeStateOPEN,
+			},
+			expectedStatus: models.SISAInitialRequestFormInProgress,
+			expectError:    true,
+		},
+	}
+	systemIntakeAdminStatusRunTestCollection(t, intakeFormTests, "Testing all system intake form Statuses")
+	systemIntakeAdminStatusRunTestCollection(t, draftBizCaseTests, "Testing all draft Buisness Case Statuses")
+	systemIntakeAdminStatusRunTestCollection(t, grtMeetingTests, "Testing all GRT Meeting Statuses")
+	systemIntakeAdminStatusRunTestCollection(t, finalBizCaseTests, "Testing all final Buisness Case Statuses")
+	systemIntakeAdminStatusRunTestCollection(t, grbMeetingTests, "Testing all GRB Meeting Statuses")
+	systemIntakeAdminStatusRunTestCollection(t, decisionTests, "Testing all Decision and Next Steps Statuses")
+
+}
+
+// systemIntakeAdminStatusRunTestCollection is a helper function to run all listed sub test cases
+func systemIntakeAdminStatusRunTestCollection(t *testing.T, tests []testSystemIntakeAdminStatusType, testType string) {
+	t.Run(testType, func(t *testing.T) {
+		for _, test := range tests {
+			t.Run(test.testCase, func(t *testing.T) {
+				verifySystemIntakeAdminStatusView(t, test)
+			})
+		}
+	})
+}
+
+// verifySystemIntakeAdminStatusView is a helper function to validate the logic for a specific test case
+func verifySystemIntakeAdminStatusView(t *testing.T, test testSystemIntakeAdminStatusType) {
```

Comment:
Stylistic nit: I'd probably inline this into where it's called in `systemIntakeAdminStatusRunTestCollection()`; the logic is pretty simple, I don't think it needs to be broken out into a separate function, and having it inlined makes it a bit easier to understand at first glance.

Reply:
That's fine with me, I'll go ahead and update it.